zebra/zebra_rib.c

```diff
@@ -209,9 +209,10 @@ route_entry_copy_nexthops (struct route_entry *re, struct nexthop *nh)
   if (nh->nh_label)
     nexthop_add_labels (nexthop, nh->nh_label_type, nh->nh_label->num_labels,
                         &nh->nh_label->label[0]);
+  nexthop->rparent = NULL;
   route_entry_nexthop_add(re, nexthop);
   if (CHECK_FLAG(nh->flags, NEXTHOP_FLAG_RECURSIVE))
-    copy_nexthops(&nexthop->resolved, nh->resolved);
+    copy_nexthops(&nexthop->resolved, nh->resolved, nexthop);
```

Comment:
This seems awfully similar to `copy_nexthops` to me. Couldn't it just be something like:
<pre>assert(!re->nexthop);
copy_nexthops(&re->nexthop, nh, NULL);
re->nexthop_num++;</pre>
Otherwise, same as above: Is it correct that `rmap_src` is not copied?